tests/doc_scan_service/session/create/session.specification.builder.test.js

```diff
@@ -204,4 +212,59 @@ describe('SessionSpecificationBuilder', () => {
 
     expect(JSON.stringify(sessionSpec)).toBe(expectedJson);
   });
+
+  it('should build SessionSpecification with subject when identityProfileRequirements exists', () => {
+    const identityProfileRequirementsDescriptor = {
+      trust_framework: 'UK_TFIDA',
+      scheme: {
+        type: 'DBS',
+        objective: 'STANDARD',
+      },
+    };
+
+    const subjectDescriptor = {
+      subject_id: 'some_subject_id_string',
+    };
+
+    const sessionSpec = new SessionSpecificationBuilder()
+      .withIdentityProfileRequirements(identityProfileRequirementsDescriptor)
+      .withSubject(subjectDescriptor)
+      .build();
+
+    const expectedJson = JSON.stringify({
+      requested_checks: [],
+      requested_tasks: [],
+      required_documents: [],
+      identity_profile_requirements: {
+        trust_framework: 'UK_TFIDA',
+        scheme: {
+          type: 'DBS',
+          objective: 'STANDARD',
+        },
+      },
+      subject: {
+        subject_id: 'some_subject_id_string',
+      },
+    });
+
+    expect(JSON.stringify(sessionSpec)).toBe(expectedJson);
+  });
+
+  it('should not build SessionSpecification with subject if identityProfile is not present', () => {
+    const subjectDescriptor = {
+      subject_id: 'some_subject_id_string',
+    };
+
+    const sessionSpec = new SessionSpecificationBuilder()
+      .withSubject(subjectDescriptor)
+      .build();
+
+    const expectedJson = JSON.stringify({
+      requested_checks: [],
+      requested_tasks: [],
+      required_documents: [],
+    });
+
+    expect(JSON.stringify(sessionSpec)).toBe(expectedJson);
+  });
```

Comment:
Same here related to my previous comment....
We do not care here about validating that the subject is passed in the context of the `identityProfileRequirements`.

I know it sounds annoying and bizarre, but such was the decision on the others SDK... and for flexibility reason, trying to keep it loose.